### DISCUSSION TITLE
Any way to goat out an English accent in men?

### DISCUSSION THREAD

**Discussion by Cowboykitty** [neonbjb/tortoise-tts#267] [Q&A]
I have been trying to clone an Englishman's voice, but it comes out American

I have noticed the opposite with women.

I was wondering if this is due to the model's training or my sample.

**Comment by neonbjb** [comment]
Probably the models training. There were definitely a lot of british female audiobook readers in the dataset.

**Comment by sbersier** [comment]
I tried with Stephen Fry's voice (which I made with a youtube video and the get_conditioning_latents.py script:
https://drive.google.com/file/d/1VgZToukf7b3gBhZtH9Tf0YvGJ60JTsFC/view?usp=sharing

Is it british enough for you? :-)
Note:  I lost the second "Who's there?" in the (needed) editing process.

**Reply by Cowboykitty** [reply]
That is impressive! I'll guess I'll play around with those for a while!

**Reply by geocine** [reply]
What is the purpose of `get_conditioning_latents.py` and how do you use it? Do you have a quick writeup somewhere how you did this. I have only tried the basic inference script
```
gen = tts.tts_with_preset(text, voice_samples=voice_samples, conditioning_latents=conditioning_latents,  preset=preset)
```

**Reply by DamienMcE** [reply]
Same. Stephen Fry works quite well for some reason, maybe because he enunciates so clear,  Hugh Laurie always sounds to high pitch.

**Reply by omega3** [reply]
Can you share this model, please? Is the best and clearly spoken British accent I've ever heard from piper, coqui-tts and similar projects.

Is there any instruction how to achieve such result? Is it possible on average CPU, without GPU? Can this be achieved with an hour youtube video as voice input, or a folder of a hundred (or more) of short mp3? For example could I have in folder mp3 from some freely available source like BBC learn English or similar and make a correct British accent model that speak without mistakes? I am talking about correct British pronunciation that is diffrent from US, for [example](https://www.ldoceonline.com/dictionary/advanced).

**Comment by sbersier** [comment]
If you generate from audio files it will create these latents anyway and it takes time. So, by generating it once and for all you can spare you some time at inference.  The get_conditioning_latents.py is there for that reason.

**Reply by geocine** [reply]
Once the latents are generated. How do you load it for inference , which parameter to use to load it?

**Comment by ef1500** [comment]
It took me a few hours, but I finally got something repeatable that actually worked. 
It may be different in your case, because you're working with male voices, and I was working with female voices, but I think the processes might be about the same.

### TL;DR: you need more samples

It took me 132 samples of ~10-second clips of a female voice for it to _not_ come out British. It was frustrating, but I managed.
I created a batch script (It's kind of annoying, I couldn't figure out how to make it CALL properly without throwing weird errors)
```bat
@echo off

set folder=%1

for /r "%folder%" %%i in (*.mp3) do (
  echo %%~ni
  START /W ffmpeg -i "%%i" -c:a pcm_f32le -ar 22050 -f segment -segment_time 10 %%~ni_%%03d.wav
)
pause
```

This segmented my mp3 files and created the proper wav files.
I then painstakingly went through as many as I could and tried to delete the ones with minimal audio in them. Usually the ones that started with silence, and kept silence for the first five seconds I just deleted. 

After that, I ran the command to generate some text, and [here's](https://files.catbox.moe/j0qsxg.wav) the result.

Hopefully this spares you some frustration. Cheers :)

**Reply by RichieRunner** [reply]
The process you're describing can be done easily with Audacity 😉
Thanks for sharing the result. Can you also share what the sample sounded like?

**Reply by FurkanGozukara** [reply]
How is the process could you show

**Reply by DamienMcE** [reply]
Wow, 132! I thought I was going over the top with 20. Will try it with more.

**Reply by ef1500** [reply]
Seems to be that if the voice is monotone, you need more samples. The more "diverse" the voice sounds, the less samples you will need.

**Reply by SenTheNastyDynast** [reply]
Has someone done this successfully to replicate an accent? I'm tempted to try this because the only thing that this program lacks is the ability to recreate accents, if someone can attest to it I might try it myself

**Comment by sbersier** [comment]


**Reply by ef1500** [reply]
https://files.catbox.moe/aelpk0.wav Not too sure what happened....
Just seems to output a bunch of static noise. (Warning that audio clip is loud)

**Reply by ef1500** [reply]
My apologies. Came from the script. The command was
`audiosplit.py -i samples -m 5 -t 10  -b  220 -f wav`
It seemed to have amplified the audio somehow, playing around with the arguments seemed to yield a slightly better result, but nevertheless, seems that the audio is somehow being amplified.

**Comment by sbersier** [comment]


**Reply by ef1500** [reply]
There's a chunks folder created, it had one file in it, the one that I sent. It was just very loud noise.

**Comment by sbersier** [comment]


**Reply by ef1500** [reply]
Alright, sounds good. Thanks for doing your best to lend a hand. I really appreciate it.

**Reply by dekker33** [reply]
Yeah, I simply cannot get an Australian accent out of the custom voice. I've trained it on almost 2 minutes of audio and still comes out american. Has anyone got any updates on how to achieve proper accents?
